# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/win/RemoteAccessible.cpp
# Commit: 5e92b9c7d479
# Full Hash: 5e92b9c7d4794af1d7ab0fb5c278c2e681a18fc6
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2023-01-28 21:11:06
# Regressor Bug: 1811092
# File Overlap Count: 1
# Description:
#   Bug 1811092 - P1: Add new async add/set text selection message. r=Jamie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D167201
# ==============================================================================

diff -r 7cebaa85f4ff -r 5e92b9c7d479 accessible/ipc/win/RemoteAccessible.cpp
--- a/accessible/ipc/win/RemoteAccessible.cpp	Fri Jan 27 20:03:06 2023 +0000
+++ b/accessible/ipc/win/RemoteAccessible.cpp	Fri Jan 27 20:15:11 2023 +0000
@@ -676,21 +676,6 @@
   *aEndOffset = end;
 }
 
-bool RemoteAccessible::AddToSelection(int32_t aStartOffset,
-                                      int32_t aEndOffset) {
-  if (StaticPrefs::accessibility_cache_enabled_AtStartup()) {
-    // Not yet supported by the cache.
-    return false;
-  }
-  RefPtr<IAccessibleText> acc = QueryInterface<IAccessibleText>(this);
-  if (!acc) {
-    return false;
-  }
-
-  return SUCCEEDED(acc->addSelection(static_cast<long>(aStartOffset),
-                                     static_cast<long>(aEndOffset)));
-}
-
 bool RemoteAccessible::RemoveFromSelection(int32_t aSelectionNum) {
   if (StaticPrefs::accessibility_cache_enabled_AtStartup()) {
     // Not yet supported by the cache.
@@ -937,5 +922,16 @@
   return GroupPos();
 }
 
+bool RemoteAccessible::SetSelectionBoundsAt(int32_t aSelectionNum,
+                                            int32_t aStartOffset,
+                                            int32_t aEndOffset) {
+  if (!StaticPrefs::accessibility_cache_enabled_AtStartup()) {
+    return false;
+  }
+
+  return RemoteAccessibleBase<RemoteAccessible>::SetSelectionBoundsAt(
+      aSelectionNum, aStartOffset, aEndOffset);
+}
+
 }  // namespace a11y
 }  // namespace mozilla